apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -145,6 +151,23 @@ fn split_subgraph(
         .try_collect()
 }
 
+/// Filter out directives from a directive list
+fn filter_directives<'a, D, I, U>(deny_list: &IndexSet<&Name>, directives: D) -> U
```

Comment:
this is a good idea, but won't be useful in practice because extract_subgraphs_from_supergraphs doesn't care about directives not involved in query planning so there won't be anything to copy

Reply:
oh i see how this is copying over keys/requires/etc — that does make many things work in our current examples. but we'll need to control how we create keys more carefully so it'll be a bit more work, unfortunately!